src/cortxfs/cortxfs_ops.c

```diff
@@ -716,44 +729,38 @@ int cfs_rename(struct cfs_fs *cfs_fs, cfs_cred_t *cred,
 			 * will be closed.
 			 */
 			log_trace("Detaching a file from the tree "
-				  "(%llu, %llu, %s)", *dino_dir, dino, dname);
-			RC_WRAP_LABEL(rc, out, cfs_detach, cfs_fs, cred,
-				      dino_dir, &dino, dname);
+				  "(%llu, %llu, %s)", *dino_dir, *dst_ino,
+				  dname);
+			RC_WRAP_LABEL(rc, out, cfs_detach2, ddir_fh, dst_fh,
+				      cred, dname);
 		}
-	} else {
-		ino_to_node_id(dino_dir, &dnode_id);
 	}
 
+	str256_from_cstr(k_sname, sname, strlen(sname));
+	str256_from_cstr(k_dname, dname, strlen(dname));
+
 	if (rename_inplace) {
 		/* a shortcut for renaming only a dentry
 		 * without re-linking of the inodes.
 		 */
-		RC_WRAP_LABEL(rc, out, cfs_tree_rename_link, cfs_fs,
-			      sino_dir, &sino, &k_sname, &k_dname);
+		RC_WRAP_LABEL(rc, out, cfs_tree_rename_link, sdir_fh, src_fh,
+			      &k_sname, &k_dname);
 	} else {
-		RC_WRAP_LABEL(rc, out, cfs_kvnode_load, &snode, cfs_fs->kvtree,
-			      &sino);
-                RC_WRAP_LABEL(rc, out, cfs_get_stat, &snode, &stat);
-                s_mode = stat->st_mode;
-                kvs_free(kvstor, stat);
-
-		node_id_t snode_id, new_node_id;
+		s_mode = src_stat->st_mode;
 
-		ino_to_node_id(sino_dir, &snode_id);
-		ino_to_node_id(&sino, &new_node_id);
-
-		RC_WRAP_LABEL(rc, out, kvtree_detach, cfs_fs->kvtree, &snode_id,
+		RC_WRAP_LABEL(rc, out, kvtree_detach, cfs_fs->kvtree, sdir_id,
 			      &k_sname);
 
-		RC_WRAP_LABEL(rc, out, kvtree_attach, cfs_fs->kvtree, &dnode_id,
-			      &new_node_id, &k_dname);
+		RC_WRAP_LABEL(rc, out, kvtree_attach, cfs_fs->kvtree, ddir_id,
+			      src_id, &k_dname);
 
 		if (S_ISDIR(s_mode)) {
-			RC_WRAP_LABEL(rc, out, cfs_update_stat, &snode,
+			/* DOUBT: Shouldn't we update the source directory
+			 * I mean parent stat to decrement the link count
```

Comment:
Is this comment required? The next code statement decrements the link count in the source directory.

Reply:
Done